File: SampSharpGameMode/Story/Actions/Cook.cs
using SampSharp.SyntheticGameMode.Extensions;
using SyntheticVideo2language.Story.Api;
using System;
using System.Collections.Generic;
using System.Text;
using System.Threading;
using System.Threading.Tasks;

namespace SampSharp.SyntheticGameMode.Story.Actions
{
    public class Cook : StoryActionBase
    {
        public override string Description { get => " is cooking food "; set { } }
        public override IStoryActor Performer { get; set; }
        public override IStoryItem TargetItem { get; set; }

        public override async Task<bool> ApplyAsync(params object[] parameters)
        {
            SampStory.Instance.History.Add(this);

            var player = Performer as Player;
            SampStory.Instance.Logger.Log(player.Description + this.Description + " at the " + TargetItem.Description);
            player.ApplyAnimation("INT_HOUSE", "wash_up", 4.1f, true, false, false, true, 3000, true);
            Thread.Sleep(3500);
            player.ClearAnimations();
            await Task.Delay(1000);

            return await (NextLocation as Location).GetNextRandomValidAction().ApplyAsync();
        }
    }
}

File: story/Actions/Cook.lua
Cook = class(StoryActionBase, function(o, params)
    params.description = " is cooking food "
    params.name = 'Cook'
    StoryActionBase.init(o, params)
end)

function Cook:Apply()
    local story = GetStory(self.Performer)
    table.insert(story.History[self.Performer:getData('id')], self)
    StoryActionBase.Apply(self)
    
    StoryActionBase.GetLogger(self, story):Log(self.Description .. " on the " .. self.TargetItem.Description, self)

    self.Performer:setAnimation("INT_HOUSE", "wash_up", 3000, true, true, false, true)
    if DEBUG then
        outputConsole("WashHands:Apply")
    end

    OnGlobalActionFinished(3000, self.Performer:getData('id'), self.Performer:getData('storyId'))
end

function Cook:GetDynamicString()
    return 'return Cook{}'
end